security/security.manager.ts

```diff
@@ -0,0 +1,111 @@
+
+  constructor() {
+    this.service = new SecurityService();
+  }
+
+  async initialization(iconikService: IconikService) {
+    try {
+      const apiUrl = getEnv('CHANGE_REFRESH_TOKEN_PERIOD_CA_URL');
+
+      const field: MetadataFieldSchema = await iconikService.metadata.createMetadataField(refreshHoursField);
+      const view = await iconikService.metadata.createMetadataView({
+        name: 'win_RefreshHoursView',
+        label: 'Set Token Refresh Time',
+        view_fields: [field],
+      });
+      await iconikService.metadata.addCategoryToView(view.id, 'custom_actions');
+
+      const customAction: CustomActionSchema = await iconikService.assets.createCustomAction('NONE', {
+        context: 'NONE',
+        type: 'POST',
+        title: 'Change Refresh Token Period',
+        url: apiUrl,
+        metadata_view: view.id,
+      });
+
+      await this.service.removeAclFromCA(iconikService, customAction);
+
+      return { message: 'Security Workflow successfully initialize.' };
+    } catch (error) {
+      console.log('initialize error', error);
+      throw new HttpBadRequestError('Cannot initialize Security Workflow. Connect WIN Support team.');
+    }
+  }
+
+  async changeRefreshTokenPeriod(
+    refreshHours: string,
+    refreshTokensLambdaARN: string,
+    invalidateTokensLambdaARN: string
+  ): Promise<{ message: string }> {
+    const refreshTokensRuleName = 'refresh-tokens-event';
+    const tokensRefreshTime = parseFloat(refreshHours) * 60;
+
+    const invalidateTokensRuleName = 'invalidate-tokens-event';
+    const tokensInvalidateTime = tokensRefreshTime + parseFloat(getEnv('INVALIDATE_TOKEN_DELAY_HOURS')) * 60;
+
+    await this.service.createRuleAndBindLambda(
+      invalidateTokensRuleName,
+      tokensInvalidateTime,
+      invalidateTokensLambdaARN
+    );
+
+    /** We have to create refreshTokensRule
+     *  after invalidateTokensRule for token
+     *  replacement to work correctly */
+    await new Promise((res) => setTimeout(() => res(), 5000));
+
+    await this.service.createRuleAndBindLambda(refreshTokensRuleName, tokensRefreshTime, refreshTokensLambdaARN);
+
+    return { message: 'Tokens refresh time changed successfully.' };
+  }
+
+  async refreshTokens(iconikService: IconikService): Promise<{ message: string }> {
+    const customActions: CustomActionSchema[] = await this.service.getCustomActions(iconikService);
+    const webHooks: WebhookResponseSchema[] = await this.service.getWebHooks(iconikService);
+
+    const invalidationTokens: IconikTokenSchema[] = this.service.getTokensFromWHandCA(webHooks, customActions);
+    if (invalidationTokens?.length) {
+      const batchPut = await IconikTokenModel.batchPut(invalidationTokens);
+      console.log('batchPut', batchPut);
+    }
+
+    const newToken: string = await this.service.createNewAppToken(iconikService);
+    await this.service.refreshTokensInWHandCA(iconikService, webHooks, customActions, newToken);

```

Comment:
I see the issue here. I set `auth-token` header for ALL CAs and WHs. There are may be stuffs that our service don`t use. Better to find way to update only our CAs and WHs. Probably, add an header like `application-name: ${STACK-NAME}`. And filer by that header.

Reply:
@azhivaeva-flo what do you think?

Reply:
I like the idea. Let's do it 😄

Reply:
then we need to make some rule that tells each developer to add `application-name: $ {NAME}` to CAs and WHs

Reply:
Yes, add it to README and I'll add it to our regulations later